src/frontend/Ast_to_Mir.ml

```diff
@@ -458,7 +458,8 @@ let create_decl_with_assign decl_id declc decl_type initial_value transform
             () } in
   let decl =
     Stmt.
-      { Fixed.pattern= Decl {decl_adtype; decl_id; decl_type; initialize= true}
+      { Fixed.pattern=
+          Decl {decl_adtype; decl_id; decl_type; initialize= Default}
```

Comment:
I think this should use the logic on the next line which inspects the rhs_assignment to set `Initialize`

Reply:
Looking at this code it's not modifying the params. Making changes in places that do no modify params is out of the scope of this PR.

Reply:
Sounds good -- I think we would really want to fix https://github.com/stan-dev/stanc3/issues/1295 before doing this -- probably by writing a generic-ish `check_size` function in Stan which we can call after the assignment is completed.